books/data_transformation.py

```diff
@@ -1,108 +1,128 @@
+        self.data_frames['books'] = books_frame
+
+    def transform_users(self, source_path):
+        users_frame = pd.read_csv(os.path.join(source_path, 'users.csv'), usecols=['Location', 'Age'], delimiter=';', encoding='windows-1250', low_memory=False)
+        df_location = users_frame['Location'].str.split(',', expand=True)
+        df_location = df_location.drop(columns=[1, 3, 4, 5, 6, 7, 8])
+
+        users_frame = users_frame.drop(columns=['Location'])
+        users_frame = pd.concat([users_frame, df_location], axis=1)
+        new_users_column_names = {
+            'Age': 'user_age',
+            0: 'user_city',
+            2: 'user_country'
+        }
+
+        users_frame.rename(columns=new_users_column_names, inplace=True)
+        users_frame['user_city'] = users_frame['user_city'].str.title()
+        users_frame['user_country'] = users_frame['user_country'].str.title()
+        users_frame['user_age'] = users_frame['user_age'].fillna(0).astype(int)
+        self.data_frames['users'] = users_frame
+
+    def transform_ratings(self, source_path):
+        ratings_frame = pd.read_csv(os.path.join(source_path, 'ratings.csv'), usecols=['ISBN', 'Book-Rating'], delimiter=';', encoding='windows-1250', low_memory=False)
+        new_ratings_column_names = {
+            'ISBN': 'r_book_isbn',
+            'Book-Rating': 'r_book_rating',
+        }
+        ratings_frame.rename(columns=new_ratings_column_names, inplace=True)
+        self.data_frames['ratings'] = ratings_frame
+
+    def transform_user_api_data(self, source_path):
+        with open(os.path.join(source_path, 'users_api_data.json'), 'r', encoding='utf-8') as f:
+            users_api_data = json.load(f)
+        users = users_api_data.get('results', [])
+        df_users = pd.json_normalize(users)
+        df_users.reset_index(inplace=True)
+        df_users = df_users.drop(columns=[
+            'index', 'gender', 'email', 'phone', 'cell', 'nat', 'name.title',
+            'name.first', 'name.last', 'location.street.number',
+            'location.street.name', 'location.state', 'location.postcode',
+            'location.coordinates.latitude', 'location.coordinates.longitude',
+            'location.timezone.offset', 'location.timezone.description',
+            'login.uuid', 'login.username', 'login.password', 'login.salt',
+            'login.md5', 'login.sha1', 'login.sha256', 'dob.date', 'dob.age',
+            'registered.date', 'id.name', 'id.value',
+            'picture.large', 'picture.medium', 'picture.thumbnail'
+        ])
+
+        new_api_users_column_names = {
+            'location.city': 'user_city',
+            'location.country': 'user_country',
+            'registered.age': 'user_age'
+        }
+        df_users.rename(columns=new_api_users_column_names, inplace=True)
+        df_users['user_age'] = df_users['user_age'].fillna(0).astype(int)
+        self.data_frames['api_users'] = df_users
```

Comment:
this should not be here at all. Just save the results to the target file here :)